packages/js/src/ui/components/InboxTabs/InboxTabs.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable local-rules/no-class-without-style */
+import { createMemo, createSignal, onMount, Show } from 'solid-js';
+import { Tabs } from '../primitives';
+import { NotificationList } from '../Notification';
+import { InboxTab } from './InboxTab';
+import { OptionsDropdown } from '../elements/OptionsDropdown';
+import { Check } from '../../icons';
+import { cn } from '../../helpers';
+
+type InboxTabsProps = {
+  tabs: Array<{ label: string; value: Array<string> }>;
+};
+
+export const InboxTabs = (props: InboxTabsProps) => {
+  const [activeTab, setActiveTab] = createSignal<string>((props.tabs[0] && props.tabs[0].label) ?? '');
+  const [tabsList, setTabsList] = createSignal<HTMLDivElement>();
+  const [visibleTabs, setVisibleTabs] = createSignal<InboxTabsProps['tabs']>([]);
+  const [dropdownTabs, setDropdownTabs] = createSignal<InboxTabsProps['tabs']>([]);
+
+  onMount(() => {
+    const tabsListEl = tabsList();
+    if (!tabsListEl) return;
```

Comment:
I think `tabsListEl` is set here https://docs.solidjs.com/concepts/refs#refs-in-solid

Reply:
I think there's not need to check for it being undefined

Reply:
the type is `HTMLDivElement | undefined` so I am forced to do that ;)